src/ast/TAGS.ts

```diff
@@ -1,13 +1,18 @@
 import NODE from "./NODE";
+import TAG from "./TAG";
 
 export default class TAGS extends NODE {
-  parse() {
-    // stub
-    throw new Error("Not implemented");
-  }
+  tags: TAG[] = [];
 
-  evaluate() {
-    // stub
-    throw new Error("Not implemented");
+  parse() {
+    this.tokenizer.getAndCheckToken("Tags:");
+    while (this.tokenizer.moreTokens()) {
+      let tag = new TAG();
+      tag.parse();
+      this.tags.push(tag);
+      if (this.tokenizer.checkNext() === ",") {
+        this.tokenizer.getNext();
+      }
+    }
```

Comment:
To handle cases where user's input is one of: Tags: language ,, midterm | Tags: ,language
we should filter out the commas (same as DECKS)
```suggestion
    while (this.tokenizer.moreTokens()) {
      let tag = new TAG();
      tag.parse();
      this.tags.push(tag);
    }
    tags.filter((tag) => {
        return tag.tagName !== ",";
    });
```

Reply:
Addressed in the other comment